scripts/configurator/parser/cmake_parser.py

```diff
@@ -0,0 +1,37 @@
+from cmake_row import cmake_row, configurator_type, row_type
+from typing import List
+
+def read_file(filepath):
+   with open(filepath) as f:
+      return f.readlines()
+
+def lines_to_rows(file_lines: List[str]) -> List[cmake_row]:
+   """
+   turns the lines of a file (parsed previously) to a list of cmake_rows,
+   skipping internal variables.
+   """
+   cmake_rows: List[cmake_row] =  []
+   comment = ""
+
+   for i, line in enumerate(file_lines):
+      row = cmake_row(line, i, comment)
+      if row.row_type == row_type.SLASH_COMMENT:
+         comment += row.val
+      elif row.row_type == row_type.VARIABLE and \
+            row.configurator_type != configurator_type.INTERNAL:
+         cmake_rows.append(row)
+         comment = ""
+
+      return cmake_rows
+
+def rows_to_lines(file_lines: List[str],
+                  cmake_rows: List[cmake_row]) -> List[str]:
+   for row in cmake_rows:
+      if row.row_type == row_type.VARIABLE:
+         file_lines[row.row_number] = row.serialize()
```

Comment:
**NOPE!** Yesterday I wrote the pseudo code just to avoid this mistake once again:
I'm copy-pasting this for the last time. The with the rows + variables approach, the serialize should look like this:
```
for each row in rows:
    fd.write(row.serialize() + "\n")
```

We talked about both the approaches. If you want to keep the rows, you discard the file lines. You have **only** rows now! The the alternative is that you have only file lines and variables, but no rows. In that case you iterate over all the variables and replace the right lines with the newly serialized value.

It should be simple: 
de-serialize: input=lines, output=cmake_rows
serialize: input=cmake_rows, output=lines

The row.serialize() method should know what to do in case the row contains a variable and call self.var.serialize() or something like that.